service/dap/server.go

```diff
@@ -1707,9 +1711,24 @@ func (s *Session) getMatchingBreakpoints(prefix string) map[string]*api.Breakpoi
 }
 
 func (s *Session) onSetExceptionBreakpointsRequest(request *dap.SetExceptionBreakpointsRequest) {
-	// Unlike what DAP documentation claims, this request is always sent
-	// even though we specified no filters at initialization. Handle as no-op.
-	s.send(&dap.SetExceptionBreakpointsResponse{Response: *newResponse(request.Request)})
+	enabled := make(map[string]bool)
+	for _, filter := range request.Arguments.Filters {
+		enabled[filter] = true
+	}
+	for _, bp := range s.debugger.Breakpoints(false) {
+		if bp.ID < 0 && enabled[bp.Name] != !bp.Disabled {
+			bp.Disabled = !enabled[bp.Name]
+			s.debugger.AmendBreakpoint(bp)
+		}
+	}
+	resp := &dap.SetExceptionBreakpointsResponse{Response: *newResponse(request.Request)}
+	for _, bp := range s.debugger.Breakpoints(false) {
```

Comment:
I think this can be folded into the loop above right? Move the creation of `SetExceptionBreakpointsResponse` above the loop, then after amend add to the `resp.Body.Breakpoints` and update the response.

Reply:
It doesn't work (that's how I did it at first) because AmendBreakpoint doesn't update the list of addresses in its argument. I did it this way because to make it work with a single loop we'd have to add a loop over all breakpoints to amendBreakpoint to collect addresses, like convertBreakpoint does, work that in most cases will be discarded (because AmendBreakpoint doesn't send the breakpoint back).